app/models/rbac.rb

```diff
@@ -248,7 +248,33 @@ def self.find_targets_with_user_group_rbac(klass, scope, rbac_filters, find_opti
     end
   end
 
+  def self.group(user_or_group)
+    case user_or_group
+    when User
+      user_or_group.current_group
+    when MiqGroup
+      user_or_group
+    when NilClass
+    else
+      raise
+    end
+  end
+
+  def self.find_options_for_tenant(klass, user_or_group, find_options = {})
+    tenant_id = group(user_or_group).try(:tenant_id)
+    return find_options unless tenant_id
+
+    tenant_id_clause = ["#{klass.table_name}.tenant_id = ? OR #{klass.table_name}.tenant_id IS NULL", tenant_id]
+    find_options[:conditions] = MiqExpression.merge_where_clauses(find_options[:conditions], tenant_id_clause)
+    find_options
+  end
+
   def self.find_targets_with_rbac(klass, scope, rbac_filters, find_options = {}, user_or_group = nil)
+    if klass.scope_by_tenant?
```

Comment:
Yeah, I'm ok with explicitly defining witch classes opt in for tenancy scoping. I would actually make it easier to phase in classes as we add support for assigning and obtaining their tenants. But, I think it would still be better to define that in RBAC instead of the individual classes. I know it seems counter to the way you would normally do it but I think in this case RBAC should be the one to dictate which classes it is going to enforce RBAC on.